cmd/chart/upgrade.go

```diff
@@ -83,38 +86,47 @@ Otherwise, it returns a non-zero exit code and the updated values.yaml file.`,
 			}
 		}
 
-		updated := 0
-		for k := range filtered {
-
-			imageName, tag := splitImageName(k)
-			ref, err := crane.ListTags(imageName)
-			if err != nil {
-				return errors.New("unable to list tags for " + imageName)
-			}
-
-			var vs []*semver.Version
-			for _, r := range ref {
-				v, err := semver.NewVersion(r)
-				if err == nil {
-					vs = append(vs, v)
+		wg := sync.WaitGroup{}
+		wg.Add(workers)
+
+		workChan := make(chan string, workers)
+		errChan := make(chan error, workers)
+		var mu sync.Mutex
+
+		updatedCount := 0
+
+		for i := 0; i < workers; i++ {
+			go func() {
+				defer wg.Done()
+				for image := range workChan {
+					if len(image) > 0 {
+						updated, imageNameAndTag, err := updateImages(image, verbose)
+						if err != nil {
+							errChan <- err
+							continue
+						}
+						if updated {
+							mu.Lock()
+							updatedCount++
```

Comment:
What are your thoughts on having a second map that starts out empty, and gets populated as images are updated? In that way, there'd be no need for updatedCount. I don't think you'd need to guard the destination map with a mutex either?